imports/plugins/included/taxes-taxcloud/server/hooks/hooks.js

```diff
@@ -1,8 +1,182 @@
 import { Meteor } from "meteor/meteor";
 import { HTTP } from "meteor/http";
-import { Logger, MethodHooks } from "/server/api";
-import { Shops, Cart, Packages } from "/lib/collections";
+import { Logger, MethodHooks, Reaction } from "/server/api";
+import { Shops, Cart } from "/lib/collections";
 
+/**
+ * @name zipSplit
+ * @summary returns ZIP4 and ZIP4 value from a zip code.
+ * @method
+ * @param  {String} pincode the pincode to be parsed
+ * @return {Object} returns { zip4: <ZIP4>, zip5: <ZIP5> }
+ */
+function zipSplit(pincode) {
+  const zipReg = /(?:([\d]{4})-?)?([\d]{5})$/;
+  if (!pincode) return { zip4: null, zip5: null };
+  const result = pincode.match(zipReg);
+  if (!result) return { zip4: null, zip5: null };
+  return { zip4: result[1], zip5: result[2] };
+}
+
+/**
+ * @name calculateTax
+ * @summary gets tax from the API and updates the cart.
+ * @method
+ * @param  {Object} pkgSettings tax-cloud settings
+ * @param  {Object} cartToCalc cart object from the database.
+ */
+function calculateTax(pkgSettings, cartToCalc) {
```

Comment:
This function is over 100+ lines which makes it hard to follow/test. I wonder if we could try to break this down into more digestable chunks.

Reply:
Doing so.